dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -2214,6 +2223,57 @@ private static String replaceDateWithFormat(String query, String regExp) {
             return query;
         }
 
+        /**
+         *
+         * @param query
+         * @param regExp
+         * @param timeFormat
+         * @return
+         */
+        private static String replaceTimeWithFormat(String query, String regExp, String timeFormat) {
+            List<RegExMatch> matches = RegEX.find(query, regExp);
+            String originalDate;
+            String luceneDate;
+            StringBuilder newQuery;
+            int begin;
+            if ((matches != null) && (0 < matches.size())) {
+                newQuery = new StringBuilder(query.length() * 2);
+                begin = 0;
+                for (RegExMatch regExMatch : matches) {
+                    originalDate = regExMatch.getMatch();
+
+                    luceneDate = toLuceneTimeWithFormat(originalDate, timeFormat);
+
+                    newQuery.append(query.substring(begin, regExMatch.getBegin()) + luceneDate);
+                    begin = regExMatch.getEnd();
+                }
+
+                return newQuery.append(query.substring(begin)).toString();
+            }
+
+            return query;
+        }
+
+        /**
+         *
+         * @param dateString
+         * @param format
+         * @return
+         */
+        private static String toLuceneTimeWithFormat(String dateString, String format) {
+            try {
+                if (!UtilMethods.isSet(dateString))
+                    return "";
+
+                SimpleDateFormat sdf = new SimpleDateFormat(format);
```

Comment:
you can use a cache + ThreadSafeSimpleDateFormat to avoid parsing the pattern everytime

Reply:
Just in case, I meant by cache the Map<String, ThreadSafeSimpleDateFormat> dateFormatMap = new ConcurrentMap<>